src/transxchange/naptan.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright 2017 Kisio Digital and/or its affiliates.
+//
+// This program is free software: you can redistribute it and/or
+// modify it under the terms of the GNU General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see
+// <http://www.gnu.org/licenses/>.
+
+//! Module to help parsing and reading NaPTAN files
+//! https://en.wikipedia.org/wiki/NaPTAN
+
+use crate::{
+    collection::CollectionWithId,
+    model::Collections,
+    objects::{Coord, StopArea as NTMStopArea, StopPoint as NTMStopPoint},
+    Result, WGS84Coordinates,
+};
+use failure::{bail, format_err, ResultExt};
+use geo_types::Point;
+use log::info;
+#[cfg(feature = "proj")]
+use proj::Proj;
+use serde::Deserialize;
+use std::{collections::HashMap, fs::File, io::Read, path::Path};
+use zip::ZipArchive;
+
+#[derive(Debug, Deserialize)]
+pub struct Stop {
+    #[serde(rename = "ATCOCode")]
+    atco_code: String,
+    #[serde(rename = "CommonName")]
+    name: String,
+    #[serde(rename = "Longitude")]
+    longitude: f64,
+    #[serde(rename = "Latitude")]
+    latitude: f64,
+    #[serde(rename = "Indicator")]
+    indicator: String,
+}
+
+impl WGS84Coordinates for Stop {
+    fn coord(&self) -> Result<Coord> {
+        Ok(Coord {
+            lon: self.longitude,
+            lat: self.latitude,
+        })
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct StopInArea {
+    #[serde(rename = "AtcoCode")]
+    atco_code: String,
+    #[serde(rename = "StopAreaCode")]
+    stop_area_code: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct StopArea {
+    #[serde(rename = "StopAreaCode")]
+    stop_area_code: String,
+    #[serde(rename = "Name")]
+    name: String,
+    #[serde(rename = "Easting")]
+    easting: f64,
+    #[serde(rename = "Northing")]
+    northing: f64,
+}
+
+const EPSG_27700: &str = "EPSG:27700";
+// FIXME: String 'EPSG:4326' is failing at runtime (string below is equivalent but works)
+const EPSG_4326: &str = "+proj=longlat +datum=WGS84 +no_defs"; // See https://epsg.io/4326
```

Comment:
what's the need to put this as const?

Reply:
It won't change so I thought `const` was a good idea... Note that it will avoid do declare the variable in the function every time we call the function `coord()` (even if the compiler probably do the optimization...)